lib/JSON/Validator.pm

```diff
@@ -827,7 +827,7 @@ sub _validate_type_number {
   unless (_is_number($value)) {
     return E $path, "Expected $expected - got string."
       if !$self->{coerce}{numbers}
-      or !looks_like_number($value);
+      or !looks_like_number($value) or $value =~ /inf|nan/i;
```

Comment:
We could probably adapt upstream's improved check at https://metacpan.org/release/JSON-Validator/source/lib/JSON/Validator.pm#L866 (https://github.com/mojolicious/json-validator/commit/a588244bdd42e62ec890d0f646eb013a3e1494be)

```suggestion
      or $value !~ /^-?(?:0|[1-9]\d*)(?:\.\d+)?(?:[eE][+-]?\d+)?$/;
```

Reply:
@zakame Thanks. I updated it to upstream's improved check. 
I had to do 2 additional things
- Update test 't/jv-number.t` by copying it from upstream. as it was failing.
- It seems the regex 
```
/^-?(?:0|[1-9]\d*)(?:\.\d+)?(?:[eE][+-]?\d+)?$/
``` 
does not support `+` prefixed to a number (reference https://github.com/regentmarkets/bom-websocket-tests/blob/master/v3/pricing/proposals/10_proposal.t#L47) so i updated the regex to 
```
/^[+-]?(?:0|[1-9]\d*)(?:\.\d+)?(?:[eE][+-]?\d+)?$/
```

Reply:
Thanks, that looks even better! :tada: